AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.m

```diff
@@ -201,49 +201,52 @@ - (void)removeAuthToken:(nullable NSString *)authToken {
   if (self.authTokenHistoryArray != nil) {
     return (NSArray<MSAuthTokenInfo *> *)self.authTokenHistoryArray;
   }
-  NSData *encryptedData = [MS_USER_DEFAULTS objectForKey:kMSAuthTokenHistoryKey];
-  if (!encryptedData) {
-    MSLogInfo([MSAppCenter logTag], @"No token history was found.");
-    return [self getEmptyAuthTokenHistoryArray];
-  }
-  NSData *decryptedData = encryptedData ? [self.encrypter decryptData:encryptedData] : nil;
-  if (!decryptedData) {
-    MSLogWarning([MSAppCenter logTag], @"Failed to get decryptedData.");
-    return [self getEmptyAuthTokenHistoryArray];
-  }
-  NSArray<MSAuthTokenInfo *> *history = decryptedData ? [NSKeyedUnarchiver unarchiveObjectWithData:decryptedData] : nil;
+  NSArray<MSAuthTokenInfo *> *history = [self retrieveAuthTokenHistory];
   if (!history) {
-    MSLogWarning([MSAppCenter logTag], @"Failed to get unarchive decryptedData.");
-    return [self getEmptyAuthTokenHistoryArray];
+    history = [NSArray<MSAuthTokenInfo *> new];
   }
-  MSLogDebug([MSAppCenter logTag], @"Retrieved history state.");
   self.authTokenHistoryArray = history;
   return (NSArray<MSAuthTokenInfo *> *)self.authTokenHistoryArray;
 }
 
-- (NSArray<MSAuthTokenInfo *> *)getEmptyAuthTokenHistoryArray {
-  self.authTokenHistoryArray = [NSArray<MSAuthTokenInfo *> new];
-  return (NSArray<MSAuthTokenInfo *> *)self.authTokenHistoryArray;
+- (NSArray<MSAuthTokenInfo *> *)retrieveAuthTokenHistory {
+  NSData *encryptedData = [MS_USER_DEFAULTS objectForKey:kMSAuthTokenHistoryKey];
+  if (!encryptedData) {
+    MSLogInfo([MSAppCenter logTag], @"No auth token history was found.");
+    return nil;
+  }
+  NSData *decryptedData = [self.encrypter decryptData:encryptedData];
+  if (!decryptedData) {
+    MSLogWarning([MSAppCenter logTag], @"Failed to decrypt auth token history.");
```

Comment:
Should this be an `MSLogError` instead of `MSLogWarning` as this is an error case?

Reply:
sure.